core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -34,7 +42,18 @@ const getMarketPrice = async function (
     collateralAmount: BigNumber
 ): Promise<BigNumber> {
     // convert collateral into DAI
-    const daiAmount = await convertCollateralToDaiUsingRoute(network, collateral.symbol, marketId, collateralAmount);
+    const { route, fees } = await getRouteAndGasQuote(network, collateral.symbol, collateralAmount, marketId);
+    if (!route) {
+        throw new Error(`No route found for ${collateral.symbol} to DAI`);
+    }
+    const pools = await routeToPool(network, route, collateral.symbol, fees);
```

Comment:
Here again, the pool should not be generated at the time when market price is requested (or at least should be optional in case of simple conversions like displaying transaction fees in DAI)